hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordScanner.java

```diff
@@ -281,13 +281,6 @@ private void scanInternalV1(Option<KeySpec> keySpecOpt) {
         final String instantTime = logBlock.getLogBlockHeader().get(INSTANT_TIME);
         totalLogBlocks.incrementAndGet();
         if (logBlock.isDataOrDeleteBlock()) {
-          if (this.tableVersion.lesserThan(HoodieTableVersion.EIGHT)) {
-            if (!getOrCreateCompletedInstantsTimeline().containsOrBeforeTimelineStarts(instantTime)
-                || getOrCreateInflightInstantsTimeline().containsInstant(instantTime)) {
```

Comment:
we can't just remove this constraints. 
We needed this specifically for tbl6 and while preparing RLI records. 
But for a regular MOR table reader, we can't expose uncommitted data. 

So, we might have to add additional argument to log record reader and then set it to true only when we are preparing RLI records. If not, it should be false on which case the condition check and continue logic should still be there for tbl v 6. 


can we add docs here as to why we are allowing here.

Reply:
Fixed the usage. Now these checks are not made for RLI.